modules/table/src/main/java/org/apache/ignite/internal/schema/marshaller/TupleMarshallerImpl.java

```diff
@@ -187,15 +188,18 @@ private void gatherStatistics(
             }
 
             col.validate(val);
-            targetTuple.values.put(col.name(), val);
 
             if (val != null) {
+                val = MarshallerUtil.shrinkValue(val, col.type());
```

Comment:
Does ti make sense for fix-length types (true == colType.spec().fixedLength()) ?

Reply:
Done, thanks